src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,59 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    static final String DEFAULT_COLOR = (Color.WHITE).name();
+    static final int DEFAULT_RADIUS = 10;
+    private int range = 100;
+    private ColorSupplier randomColor = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int choice = random.nextInt(4);
+        switch (choice) {
+            case 0:
+                return getRandomSquare();
+            case 1:
+                return getRandomRectangle();
+            case 2:
+                return getRandomRightTriangle();
+            case 3:
+                return getRandomIsoscelesTrapezoid();
+            default:
+                throw new IllegalStateException("Incorrect value: " + choice);
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(DEFAULT_COLOR, DEFAULT_RADIUS);
+    }
+
+    private Square getRandomSquare() {
+        String color = randomColor.getRandomColor();
+        int side = random.nextInt(range);
```

Comment:
Create separate method `int randomSize()` and use it like 
`return new Rectangle(color, randomSize(), randomSize());`